chainstate/src/interface/chainstate_interface.rs

```diff
@@ -49,7 +49,17 @@ pub trait ChainstateInterface: Send + Sync {
     fn invalidate_block(&mut self, block_id: &Id<Block>) -> Result<(), ChainstateError>;
     fn reset_block_failure_flags(&mut self, block_id: &Id<Block>) -> Result<(), ChainstateError>;
     fn preliminary_block_check(&self, block: Block) -> Result<Block, ChainstateError>;
-    fn preliminary_header_check(&self, header: SignedBlockHeader) -> Result<(), ChainstateError>;
+
+    /// Check the headers. The first header's parent block must be known.
+    /// Each of the "other" headers must be connected to the previous one and the first of them
+    /// should be connected to `first_header`. These headers only get the most basic checks (e.g.
+    /// checkpoint enforcement).
+    fn preliminary_headers_check(
+        &self,
+        first_header: &SignedBlockHeader,
+        other_headers: &[SignedBlockHeader],
+    ) -> Result<(), ChainstateError>;
```

Comment:
If first_header is connected to other_headers... why aren't they all in one list?

Reply:
> If first_header is connected to other_headers... why aren't they all in one list?

Actually, initially I had them all in one list, but then decided to split them. The reasons are that, first of all, `first_header` is treated differently than the rest of the headers. Secondly, `first_header` is required, and the others are optional.
So, if you don't insist, I'd leave it as is.

Reply:
I actually kind of insist... it feels to me that this is leaking implementation details.

Besides, it's acceptable to return no errors for an empty list. It kinds of maps to "checking no blocks is not an error", unless it's required for another reason.